### PR TITLE
Default config overhaul

### PR DESCRIPTION
**Pull request by Lysandr0**
Haven't had the chance to thoroughly test it unfortunately, especially on the combat side. (Might be a bit too sensitive... Or not) 
Some values are taken directly from my own config, others have been changed based on personal experience/testing ;p 

Any opinion/suggestion is very much welcome on this one, as this request can(/may) affect people's opinion on NCP.

### CODE REVIEW

File: NCPCore/src/main/java/fr/neatmonster/nocheatplus/config/DefaultConfig.java [hunk 1]
```diff
@@ -412,31 +421,32 @@ public DefaultConfig() {
...
         set(ConfPaths.INVENTORY_INSTANTBOW_IMPROBABLE_WEIGHT, 0.6, 1085);
-        set(ConfPaths.INVENTORY_INSTANTBOW_ACTIONS, "cancel vl>15 log:instantbow:2:5:if cancel", 785);
+        set(ConfPaths.INVENTORY_INSTANTBOW_ACTIONS, "cancel vl>15 cancel log:instantbow:4:5:i vl>200 cancel log:instantbow:0:10:icf cmdc:kickbow:2:10", 1154);
 
+        // TODO: Set this to false as default?
         set(ConfPaths.INVENTORY_INSTANTEAT_CHECK, "default", 785);
         set(ConfPaths.INVENTORY_INSTANTEAT_ACTIONS, "log:instanteat:2:5:if cancel", 785);
 
         set(ConfPaths.INVENTORY_FASTCONSUME_CHECK, "default", 785);
         set(ConfPaths.INVENTORY_FASTCONSUME_DURATION, 0.7, 785);
```
Comment (xaw3ep) [inline NCPCore/src/main/java/fr/neatmonster/nocheatplus/config/DefaultConfig.java]: Time players need to consume an item is 1,5s. NCP just block if players use it lower than 0.7s so there can be bypasses. So I'm thinking about changing it to a higher value like 1.35;1.4. What do you guys think about?
PS: Most items need 1.5s to consume it, except dried_kelp(0.5s), so we should better exempt from checking it.

**Comment by TravelerJay** [comment]
I don't think I should be asking a question here. But do you think I could ask one as I haven't been able to find a place to ask. I saw the FAQ page but didn't see a comment button.

**Comment by MarkElf** [comment]
@TravelerJay Your message could have just had the question in it and I could have answered it already but you just wasted time asking if you can ask a question or not. Why would you not be able to?

**Comment by TravelerJay** [comment]
Alright I'm sorry for that. Well I just would like to know if there was a way to disable the anti cheat and just have it show if someone is hacking or not

**Comment by MarkElf** [comment]
Remove all the 'cancel' actions.

**Comment by TravelerJay** [comment]
So I would just delete all the 'cancel' actions?

**Comment by MarkElf** [comment]
That's what I said. Yes.

**Comment by TravelerJay** [comment]
Ok thank you for the help! Have a good day.

**Comment by Lysandr0** [comment]
Side note: for `FlyingFrequency` and `PacketFrequency` you may want to keep the cancel flag as that kind of packet spam can tipically harm/crash the server or even other players (paralyse/magnet hacks)

**Comment by TravelerJay** [comment]
Ok got it ill leave that in.  One last thing should be very easy to answer. Here's an example " actions: log:inventorymove:2:5:if cancel" would I delete this entire line of code because it has the word "actions:" and "cancel" in it or would I just delete the "cancel" part in it?

**Comment by Lysandr0** [comment]
You’ll only need to delete the “cancel” flag.

**Comment by TravelerJay** [comment]
Ok I got it! Thank you so much for all the help! I wish you a good night.

**Comment by Lysandr0** [comment]
To get back on topic:  unless I've overlooked something, or you guys have some more inputs/suggestions, this could now be merged to the fork.

**Comment by xaw3ep** [comment]
I haven't had time to test these config changes. This'll have to wait for a while.

**Comment by MarkElf** [comment]
import fr.neatmonster.nocheatplus.compat.BridgeMaterial;

        set(ConfPaths.BLOCKPLACE_NOSWING_EXCEPTIONS, Arrays.asList(BridgeMaterial.LILY_PAD.toString(), Material.FLINT_AND_STEEL.toString()), 1154);

Have the config spit out the correct name for lily pads when generating.

**Comment by Lysandr0** [comment]
Ty for the heads up. :p

**Comment by xaw3ep** [comment]
Just left some experiences while testing:
- set(ConfPaths.FIGHT_REACH_SURVIVALDISTANCE, 4.3, 1154); 4.3 might be for block reach. When hitting an entity, the range is somewhere within 3.1-3.5.
Not sure to give out the exact value. Maybe @MarkElf can help this
- set(ConfPaths.INVENTORY_INSTANTBOW_DELAY, 130, 785); change 130 to 75 can fix flux fastbow bypass but it looks like a bit dangerous when players have high ping or low TPS server. More testing is needed!
- set(ConfPaths.MOVING_SURVIVALFLY_LENIENCY_HBUFMAX, 1.0, 1143); This anti-false-positive system can be a disadvantage for check checking low speed modified like noslow and can be used to perform long jump bypass.
Change it to 0.5 can fix this problem but will be more likely to have false flags. Don't worry, I will make a hook to deal with those. 

Other things else seem to work pretty well. :)

**Comment by Lysandr0** [comment]
Implemented your suggestions :)
 I’ll leave the reach stuff to anyone who can actually test it more thoroughly.

**Comment by xaw3ep** [comment]
Thank you!

**Pull request merged**